src/core/Text/Text.tsx

```diff
@@ -20,6 +20,15 @@ export interface TextProps extends HtmlSpanProps {
   variant?: 'body' | 'lead' | 'bold';
 }
 
+interface InnerRef {
+  forwardedRef: React.RefObject<HTMLSpanElement>;
+}
+
+export interface TextProps extends InternalTextProps {
+  /** Ref object to be passed to the element */
+  ref?: React.RefObject<HTMLSpanElement>;
+}
+
```

Comment:
I'm wondering about these interfaces. `<HintText>` does not have these and for all intents and purposes it seems to work the same way in the end: the ref gets passed all the way to the `<span>` element. Do we need these interfaces?

Also, even though `ref` is defined here as part of the `TextProps`, it's not visible on the Styleguidist page as part of the `<Text>` component's API. I think it would be nice to have it there so developers know it can be used. Maybe also documented more specifically like: "Ref object which gets passed to the underlying span-element"

Reply:
Outdated, but in the end when forwardRef was left to be visible in the api most of these inner interfaces were removed. 
Documenting the ref is now done with forwardedRef prop.